lib/node_modules/@stdlib/lapack/base/dlaset/lib/base.js

```diff
@@ -0,0 +1,244 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isRowMajor = require( '@stdlib/ndarray/base/assert/is-row-major' );
+var loopOrder = require( '@stdlib/ndarray/base/nullary-loop-interchange-order' );
+var min = require( '@stdlib/math/base/special/fast/min' );
+
+
+// FUNCTIONS //
+
+/**
+* Initialises an M by N matrix `A` to `beta` on the diagonals and `alpha` on the off diagonals, setting all of matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in matrix `A`
+* @param {NonNegativeInteger} N - number of columns in matrix `A`
+* @param {Float64Array} A - input matrix
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index for `A`
+* @param {number} alpha - number to set on off diagonal elements
+* @param {number} beta - number to set on diagonal elements
+* @returns {Float64Array} `A`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( 4 );
+*
+* setAll( 2, 2, A, 2, 1, 0, 0.0, 1.0 );
+* // A => <Float64Array>[ 1.0, 0.0, 0.0, 1.0 ]
+*/
+function setAll( M, N, A, strideA1, strideA2, offsetA, alpha, beta ) {
+	var da0;
+	var da1;
+	var sh;
+	var S0;
+	var S1;
+	var sa;
+	var ia;
+	var i0;
+	var i1;
+	var o;
+
+	// Resolve the loop interchange order:
+	o = loopOrder( [ M, N ], [ strideA1, strideA2 ] );
+	sh = o.sh;
+	sa = o.sx;
+
+	// Extract loop variables for purposes of loop interchange: dimensions and loop offset (pointer) increments...
+	S0 = sh[ 0 ];
+	S1 = sh[ 1 ];
+	da0 = sa[ 0 ];
+	da1 = sa[ 1 ] - ( S0*sa[0] );
+
+	// Set the pointers to the first indexed elements in the respective matrices...
+	ia = offsetA;
+
+	// Iterate over the matrix dimensions...
+	for ( i1 = 0; i1 < S1; i1++ ) {
+		for ( i0 = 0; i0 < S0; i0++ ) {
+			A[ ia ] = alpha;
+			ia += da0;
+		}
+		ia += da1;
+	}
+
+	if ( alpha !== beta ) {
+		ia = offsetA;
+		for ( i1 = 0; i1 < min( M, N ); i1++ ) {
+			A[ ia ] = beta;
+			ia += strideA1 + strideA2;
```

Comment:
`strideA1 + strideA2` is a constant. No need to recompute for each iteration. See `zlaset` and its `setDiagonal` function.